ivy/functional/frontends/numpy/indexing_routines/generating_index_arrays.py

```diff
@@ -32,7 +32,7 @@ def unravel_index(indices, shape, order="C"):
 
 @outputs_to_numpy_arrays
 def diag_indices(n, ndim=2):
-    idx = ivy.arange(n, dtype=int)
+    idx = ivy.arange(n, dtype="int64")
```

Comment:
This is not a proper way for typecasting, you are restricting data generation
you have to use an `if` statement and typecast the result to a specific data type before returning to match the dtypes of other backends

Reply:
Here, datatype doesn't matter because we are just accessing the main diagonal indices.